modules/generator/encoding_test.go

```diff
@@ -0,0 +1,39 @@
+package generator
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/tempo/pkg/tempopb"
+	v1 "github.com/grafana/tempo/pkg/tempopb/trace/v1"
+	"github.com/grafana/tempo/pkg/util/test"
+)
+
+func BenchmarkOTLPDecoder(b *testing.B) {
```

Comment:
how does the otlp decoder compare to the tempo one?

Reply:
I have added a benchmark for that decoder as well.